readalongs/static/read-along-1.3.dtd

```diff
@@ -0,0 +1,109 @@
+<!-- VERSION: 1.2 -->
+<!ELEMENT read-along (meta|text|body|div|span|anchor|silence|graphic|p|s|w)*>
+<!ATTLIST read-along
+  use-assets-folder CDATA #IMPLIED
+  href CDATA #IMPLIED
+  audio CDATA #IMPLIED
+  xml:lang CDATA #IMPLIED
+  language CDATA #IMPLIED
+  lang CDATA #IMPLIED
+  version CDATA #IMPLIED
+  xmlns CDATA #IMPLIED>
+
+<!ELEMENT text (body|div|span|anchor|silence|graphic|p|s|w)*>
+<!ATTLIST text
+  xml:lang CDATA #IMPLIED
+  lang CDATA #IMPLIED
+  fallback-langs CDATA #IMPLIED
+  id CDATA #IMPLIED>
+
+<!ELEMENT body (div|span|anchor|silence|graphic|p|s|w)*>
+<!ATTLIST body
+  xml:lang CDATA #IMPLIED
+  lang CDATA #IMPLIED
+  id CDATA #IMPLIED>
+
+<!ELEMENT anchor EMPTY>
+<!ATTLIST anchor time CDATA #REQUIRED>
+
+<!ELEMENT silence EMPTY>
+<!ATTLIST silence dur CDATA #REQUIRED>
+
+<!ELEMENT graphic EMPTY>
+<!ATTLIST graphic
+  url CDATA #REQUIRED
+  id CDATA #IMPLIED>
+
+<!ELEMENT div (#PCDATA|div|span|anchor|silence|graphic|p|s|w)*>
+<!ATTLIST div
+  xml:lang CDATA #IMPLIED
+  lang CDATA #IMPLIED
+  id CDATA #IMPLIED
+  class CDATA #IMPLIED
+  type CDATA #IMPLIED
+  do-not-align CDATA #IMPLIED
+  time CDATA #IMPLIED
+  dur CDATA #IMPLIED>
+
+<!ELEMENT span (#PCDATA|div|span|anchor|silence|graphic|p|s|w)*>
+<!ATTLIST span
+  xml:lang CDATA #IMPLIED
+  lang CDATA #IMPLIED
+  id CDATA #IMPLIED
+  class CDATA #IMPLIED
+  type CDATA #IMPLIED
+  do-not-align CDATA #IMPLIED
+  time CDATA #IMPLIED
+  dur CDATA #IMPLIED>
+
+<!ELEMENT p (#PCDATA|span|anchor|silence|s|w)*>
+<!ATTLIST p
+  xml:lang CDATA #IMPLIED
+  lang CDATA #IMPLIED
+  id CDATA #IMPLIED
+  class CDATA #IMPLIED
+  do-not-align CDATA #IMPLIED
+  time CDATA #IMPLIED
+  dur CDATA #IMPLIED>
+
+<!ELEMENT s (#PCDATA|span|anchor|silence|w)*>
+<!ATTLIST s
+  xml:lang CDATA #IMPLIED
+  lang CDATA #IMPLIED
+  id CDATA #IMPLIED
+  class CDATA #IMPLIED
+  do-not-align CDATA #IMPLIED
+  time CDATA #IMPLIED
+  dur CDATA #IMPLIED
+  annotation-id CDATA #IMPLIED
+  sentence-id CDATA #IMPLIED
+  xmlns CDATA #IMPLIED>
+
+<!ELEMENT w (#PCDATA|span|syl)*>
+<!ATTLIST w
+  xml:lang CDATA #IMPLIED
+  effective-g2p-lang CDATA #IMPLIED
+  lang CDATA #IMPLIED
+  id CDATA #IMPLIED
+  class CDATA #IMPLIED
+  do-not-align CDATA #IMPLIED
+  ARPABET CDATA #IMPLIED
```

Comment:
can we make this case insensitive?

Reply:
I was thinking the same. However, I am worried about backwards compatibility. Attributes in HTML are case insensitive but are case sensitive in XML (.readalong is a subset of XML). We need a good look at the whole pipeline before we make the switch. We use both XML and HTML parsers in the various parts of the pileline. As far as I know, only the CLIs consume the ARPABET attribute.

Reply:
Maybe an option is to put both `ARPABET` and `arpabet` is the dtd?